bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/SitemapSubscriptionService.java

```diff
@@ -225,4 +249,28 @@ private Sitemap getSitemap(String sitemapName) {
         return null;
     }
 
+    @Override
+    public void modelChanged(String modelName, EventType type) {
+
+        if (type != EventType.MODIFIED || !modelName.endsWith("." + SITEMAP_SUFFIX)) {
+            return; // we process only sitemap modifications here
+        }
+
+        String changedSitemapName = modelName.substring(0, modelName.indexOf("."));
+
+        for (Entry<String, PageChangeListener> listenerEntry : pageChangeListeners.entrySet()) {
+            String sitemapWithPage = listenerEntry.getKey();
+            String sitemapName = sitemapWithPage.substring(0, sitemapWithPage.indexOf(SITEMAP_PAGE_SEPARATOR));
```

Comment:
have a look at `getPageId(...)` and `getSitemapName(...)` methods in this class. Don't duplicate this logic - rather extract a method there which you then can use here!